ResponseTimes/LogParser.fs

```diff
@@ -4,55 +4,55 @@ open System
+type Logs = CsvProvider<"sample.csv"> 
+type Details = JsonProvider<"details.json", SampleIsList = true>
 type ResponseTimeRegex = Regex< @"(?<=Request Finished in )((?<ResponseTime>\d+)(?= milliseconds))", noMethodPrefix = true >
 
-type LogDetails = {
-    ApplicationName : string;
-    LoggerName : string;
-    RequestedURL : Uri;
-    HttpResponseStatusCode : int;
-    HttpVerb : string;
-    Message : string;
-    TransactionId : Option<Guid>;
-    ResponseTime : Option<int>;
-}
+/// Only create the matcher once rather than for every call
+let private matcher = ResponseTimeRegex().Match
+
+type LogDetails =
+    { ApplicationName : string
+      LoggerName : string
+      RequestedURL : Uri
+      HttpResponseStatusCode : int
+      HttpVerb : string
+      Message : string
+      TransactionId : Guid option
+      ResponseTime : int option }
 
 let createDetailsJson () =
-    let logs = Logs.Load("logs.csv")
+    let logs = Logs.Load "logs.csv"
     use outFile = new StreamWriter("details.json")
 
-    outFile.WriteLine(sprintf "%s" "[")
-
-    let details = 
-        logs.Rows 
-        |> Seq.take(100000)
-        |> Seq.map(fun log -> log.Details)
-        |> Seq.iter (fun detail -> outFile.WriteLine(sprintf "%s," detail))
+    outFile.WriteLine "["
 
+    logs.Rows 
+    |> Seq.take 100000
+    |> Seq.map(fun log -> log.Details)
+    |> Seq.iter(sprintf "%s," >> outFile.WriteLine)
 
-    outFile.WriteLine(sprintf "%s" "]")
+    outFile.WriteLine "]"
     outFile.Close ()
 
-let convertLogToLogDetails (log : Logs.Row) =
-    let details = Details.Parse(log.Details)
-    let responseTime = ResponseTimeRegex().Match(details.Message).ResponseTime.Value
-    {
-        ApplicationName = log.ApplicationName;
-        LoggerName = log.LoggerName;
-        RequestedURL = new Uri(details.RequestedUrl);
-        HttpResponseStatusCode = details.HttpResponseStatusCode;
-        HttpVerb = details.HttpVerb;
-        Message = details.Message;
-        TransactionId = details.TransactionId;
-        ResponseTime = match responseTime with
-                       | "" -> None
-                       | _ -> Some (Int32.Parse(responseTime))
-    }
-
-
-let parseCsv (fileName : string) =
+let convertLogToLogDetails (log:Logs.Row) =
+    let details = Details.Parse log.Details
+
+    { ApplicationName = log.ApplicationName
+      LoggerName = log.LoggerName
+      RequestedURL = Uri details.RequestedUrl
+      HttpResponseStatusCode = details.HttpResponseStatusCode
+      HttpVerb = details.HttpVerb
+      Message = details.Message
+      TransactionId = details.TransactionId
+      ResponseTime =
+        match matcher(details.Message).ResponseTime.Value with
+        | "" -> None
+        | responseTime -> Some (int responseTime) }
+
+let parseCsv (fileName:string) =
     Logs.Load(fileName).Rows
-    |> Seq.map(fun log -> convertLogToLogDetails log)
\ No newline at end of file
+    |> ParStream.ofSeq
```

Comment:
What is ParStream?  Is it a parallel stream?  I'm assuming this starts streaming in the file contents (rather than waiting for the entire file to load first) and it also processes it in parallel?